src/components/redis.rs

```diff
@@ -7,24 +7,34 @@ use deadpool_redis::{
 use super::{configuration::Redis as RedisConfig, health::Healthy};
 
 #[derive(Clone)]
-pub struct RedisComponent {
+pub struct Redis {
     redis_host: String,
     pub pool: Option<Pool>,
 }
 
-impl RedisComponent {
+#[async_trait]
+pub trait RedisComponent {
```

Comment:
I would argue this could be a `KVStoreComponent` which returns our own `Connection` trait. However I understand the use here and I think it would be over designing at this point so let's forget about this comment for now. 🙃